.github/workflows/main.yml

```diff
@@ -43,6 +43,7 @@ jobs:
         run: mix deps.get
       - name: compile
         run: | 
+          git fetch --tags
```

Comment:
Did you find this was needed here in order for the `release` job with that one action to work? Since _build and so forth is not restored in the release job, I'm not sure this is needed. Thoughts?

Reply:
Its needed to make sure at compile time `bump_version.exs` can get `git tags`. For git tag to work, we need this

Reply:
In that case I think it makes sense to move this into the  `release` job, what do you think?

Reply:
we still would need it in compilation so that we can set lib version correctly